install_config/upgrading/migrating_etcd.adoc

```diff
@@ -0,0 +1,274 @@
+| etcd-test-2:2379 | a83a3258059fee18 | 3.1.9   | 25 kB   | true      |       415 |        995 |
+| etcd-test-3:2379 | 22a9f2ddf18fee5f | 3.1.9   | 25 kB   | false     |       415 |        995 |
++------------------+------------------+---------+---------+-----------+-----------+------------+
+----
++
+If minimum and maximum of raft indexes over all etcd members differ by more than
+one unit, wait a minute and try again. Additionally, the migration should not be
+run repeatedly, as new v2 data can overwrite v3 data that has already migrated.
+
+. +++<b>Migrate and scale up etcd:</b>+++
+
+.. Stop etcd on all etcd hosts:
++
+----
+# systemctl stop etcd
+----
+
+.. The migration itself is performed by running the following command (with the
+*etcd* daemon stopped) on your first etcd host:
++
+----
+# ETCDCTL_API=3 etcdctl migrate --data-dir=/var/lib/etcd
+----
++
+The `--data-dir` can point to different location depending on the deployment.
+For example, embedded etcd operates over the
+*_/var/lib/origin/openshift.local.etcd_* directory, and etcd run as a system
+container operates over the *_/var/lib/etcd/etcd.etcd_* directory.
++
+When complete, the migration responds with the following message if successful:
++
+----
+finished transforming keys
+----
++
+If there is no v2 data, it responds with:
++
+----
+no v2 keys to migrate
+----
+
+.. If the `etcdctl migrate` command is run without the `--no-ttl` option, TLL keys
+are migrated as well. Given that the TTL keys in v2 data are replaced with leases in
+v3 data, you must attach leases to all migrated TTL keys by running the following (with
+*etcd* daemon running):
+
+... Create a new cluster on the first host:
++
+----
+# echo "ETCD_FORCE_NEW_CLUSTER=true" >> /etc/etcd/etcd.conf
+# systemctl start etcd
+# sed -i '/ETCD_FORCE_NEW_CLUSTER=true/d' /etc/etcd/etcd.conf
+# systemctl restart etcd
+----
+
+... Scale up additional etcd hosts by following the
+xref:/backup_restore.adoc#adding-addtl-etcd-members[Adding Additional etcd
+Members] documentation.
+
+... After your etcd cluster is back online with all members, re-introduce the TTL
+information by running the following on the first master:
++
+----
+$ oadm migrate etcd-ttl <certificate_details> <endpoint> \
+    --ttl-keys-prefix '/kubernetes.io/events' \
+    --lease-duration 1h
+
+$ oadm migrate etcd-ttl <certificate_details> <endpoint> \
+    --ttl-keys-prefix '/openshift.io/oauth/accesstokens' \
+    --lease-duration 86400s
+$ oadm migrate etcd-ttl <certificate_details> <endpoint> \
+    --ttl-keys-prefix '/openshift.io/oauth/authorizetokens' \
+    --lease-duration 500s
+----
+
+. +++<b>Reconfigure the master:</b>+++
+
+.. After the migration is complete, the
+xref:../install_config/master_node_configuration.adoc#master-configuration-files[master
+configuration file] (the *_/etc/origin/master/master-config.yaml_* file by
+default), to be updated so the master daemons can use the new storage back end:
```

Comment:
This sounds wrong:
"After the migration is complete, the master configuration file (the *_/etc/origin/master/master-config.yaml_* file by
+default), to be updated so the master daemons can use the new storage back end"

Reply:
Whoops, fixed.